silver/api/serializers.py

```diff
@@ -378,3 +385,80 @@ def validate(self, data):
                   " Use the corresponding endpoint to update the state."
             raise serializers.ValidationError(msg)
         return data
+
+
+class PaymentUrl(serializers.HyperlinkedRelatedField):
+    def get_url(self, obj, view_name, request, format):
+        kwargs = {'customer_pk': obj.customer.pk, 'payment_pk': obj.pk}
+        return reverse(view_name, kwargs=kwargs, request=request, format=format)
+
+    def get_object(self, view_name, view_args, view_kwargs):
+        return self.queryset.get(pk=view_kwargs['payment_pk'])
+
+
+class PaymentSerializer(serializers.HyperlinkedModelSerializer):
+    url = PaymentUrl(view_name='payment-detail', source='*',
+                     read_only=True)
+
+    class Meta:
+        model = Payment
+        fields = ('id', 'url', 'customer', 'provider', 'amount', 'currency',
+                  'due_date', 'status', 'visible', 'proforma', 'invoice')
+
+    def validate(self, attrs):
+        if self.instance and self.instance.status in Payment.Status.FinalStatuses:
+            message = "Cannot update a payment with '{}' status.".format(
+                self.instance.status
+            )
+            raise serializers.ValidationError(message)
+
+        # Run model clean and handle ValidationErrors
+        try:
+            # Use the existing instance to avoid unique field errors
```

Comment:
I'm getting unique constraint errors for some unique foreign keys when trying like that...